stdlib/public/Concurrency/DispatchGlobalExecutor.cpp

```diff
@@ -314,17 +314,10 @@ clock_and_value_to_time(int clock, long long sec, long long nsec) {
   case swift_clock_id_continuous:
     return value | DISPATCH_UP_OR_MONOTONIC_TIME_MASK;
   case swift_clock_id_wall: {
-#if defined(_WIN32)
     struct timespec ts = { 
-      .tv_sec = sec,
+      .tv_sec = static_cast<time_t>(sec),
```

Comment:
Hmm, doesn't this silently truncate? I do like the idea of merging the paths.

Reply:
time_t should be 64 bits on macos. This is why teh compiler was not complaining on the implicit cast. but yes it will truncate on Android, which i am assuming is ok for this use case.

Reply:
Wouldn't that move the time backwards then?

Reply:
yeah. i think it would truncate. but we were truncating on windows also before. @phausler, @ktoso  and @grynspan thoughts?